pages/api/group/[gid]/index.ts

```diff
@@ -0,0 +1,69 @@
+import mongoose from 'mongoose';
+import type { NextApiRequest, NextApiResponse } from 'next';
+import {dbConnect} from '@/lib/mongoClient';
+import { responseFormatter } from '@/lib/responseLib';
+import Group from '@/models/Group';
+import  UserGroupRelation  from '@/models/UserGroupRelation';
+
+export default async function gidHandler(
+  req: NextApiRequest,
+  res: NextApiResponse
+) {
+
+  // Connect with MongoDB database
+  try {
+    await dbConnect();
+  } catch (e) {
+    console.log('fail to connect with database', e);
+    return (
+      res
+        .status(500)
+        .json(responseFormatter(false, null, 'Internal Server Error'))
+    );
+  }
+  const {gid} = req.query
+  const group = await Group.findOne({gid: gid});
+
+  switch (req.method) {
```

Comment:
Also, though we have middleware, but more specific authorization should be handled by each API endpoint. Once you started the front end ticket, you will probably come back and refactor. For example, can a normal member change the group information? That Authorization should be reflected in your API Endpoint like this one. Furthermore, this is when you wanna extract this logic into separate function, (wish that was Express, then it could be a middleware, but NextJS no). Because you will be using this authorization logic more than once in other endpoints.